packages/SwingSet/src/liveslots/vatstore-usage.md

```diff
@@ -0,0 +1,128 @@
+
+Durable Kinds are defined just like virtual Kinds, but they use a different constructor (`defineDurableKind` instead of `defineKind`), which requires a "handle" created by `makeKindHandle`. Durable virtual objects can only hold durable data in their `state`.
+
+The handle is a durable virtual object of a special internal Kind.
+
+# Virtual Collections
+
+Liveslots provides a handful of "virtual collection" types to vats, to store high-cardinality data on disk rather than in RAM. The collection types are accessed through the `VatStore` global, which currently has the following collection constructor functions:
+
+* `makeScalarBigMapStore`
+* `makeScalarBigWeakMapStore`
+* `makeScalarBigSetStore`
+* `makeScalarBigWeakSetStore`
+
+Each function accepts a `isDurable` argument, so there are currently 8 collection types.
+
+Each collection is assigned a Kind index. These index values are stored in `vs.storeKindIDTable`, as a mapping from the collection type name (`scalarMapStore`, `scalarDurableMapStore`, `scalarWeakSetStore`, etc) to the integer of their ID. The current value is:
+
+* `v6.vs.storeKindIDTable` : `{"scalarMapStore":1,"scalarWeakMapStore":2,"scalarSetStore":3,"scalarWeakSetStore":4,"scalarDurableMapStore":5,"scalarDurableWeakMapStore":6,"scalarDurableSetStore":7,"scalarDurableWeakSetStore":8}`
+
+which means `o+1` is the Kind ID for `scalarMapStore`.
+
+Each new store, regardless of type, is allocated the next available collection ID. This is an integer that starts at "1", and is independent of the numberspace used by exported Remotables and Kind IDs. `o+1/1` is allocated for the "baggage" Kind, indicating that it is a `scalarMapStore` (`o+1` is used for that collection type), and also that it is the first collection allocated in the vat. If userspace calls `makeScalarBigWeakSetStore()` and then `makeScalarSetStore()` at vat startup, it is likely to get `o+4/2` and `o+3/3` respectively.
+
+We examine a vat which performs the following at startup:
+
+```js
+const makeFoo = VatData.defineKind('foo',
+				   (arg) => ({ prop1: arg }),
+				   (state) => ({ doFoo: () => state.prop1 }),
+				  );
+const foo = makeFoo(1);
+const foo2 = makeFoo(2);
+const c1 = VatData.makeScalarBigMapStore('mylabel');
+c1.init('key1', foo);
+c1.init('key2', foo);
+c1.init('key3', foo);
+c1.init('key4', foo2);
+```
+
+
+Each collection stores a number of metadata keys in the vatstore, all with a prefix of `vs.vc.${collectionID}.|`. The currently defined metadata keys (copied from the record for the "mylabel" Kind stored in `c1`) are:
+
+* `v6.vs.vc.2.|entryCount`: `4`: the size of the collection (4 entries)
+* `v6.vs.vc.2.|label`:  `mylabel`: a debugging label applied when the collection is created
+* `v6.vs.vc.2.|nextOrdinal`: `1` : a counter used to allocate index values for Objects used as keys
+* `v6.vs.vc.2.|schemata`: `{"body":"[{\"@qclass\":\"tagged\",\"tag\":\"match:scalar\",\"payload\":{\"@qclass\":\"undefined\"}}]","slots":[]}`
```

Comment:
yeah, I tried to cover that up around line 93, but I'll add some more examples because it's a subtle point and I've forgotten it more than once already